ibm-mq/1.6/modules/ROOT/pages/ibm-mq-troubleshooting.adoc

```diff
@@ -0,0 +1,108 @@
+
+To troubleshoot Anypoint Connector for IBM MQ (IBM MQ Connector), become familiar with the information about enabling verbose logging, troubleshooting acknowledgment issues, troubleshooting MQ stopped consuming messages, troubleshooting redeployment errors when using binding mode, and interpreting commonly thrown messages.
+
+== Enable Verbose Logging
+
+Sometimes you are not sure why the application is failing and you require a better understanding of what is going on in the app's interaction with the MQ server.
+
+To gather detailed logs:
+
+* Use the configuration file to increase the logging level on the server side.
+
+* Enable verbose logging for JMS Connector and MQ packages:
++
+. Access Anypoint Studio and navigate to the *Package Explorer* view.
+. Open your application's project name.
+. Open the *src/main/resources* path folder.
+. Open the *log4j2.xml* file inside the folder.
+. If the following line is already in the *log4j2.xml* file, uncomment it to enable it; otherwise, add it:
++
+[source,xml,linenums]
+----
+    <AsyncLogger name=“com.mulesoft.connectors.ibmmq” level=“DEBUG”/>
+    <AsyncLogger name=“org.mule.jms.commons” level=“DEBUG”/>
+----
+
+
+== Troubleshoot Acknowledgment Issues
+
+There are several ways an application setup can be inconsistent in terms of message acknowledgment.
+You can configure the session acknowledgment for an application as:
+
+* *IMMEDIATE*
+* *AUTO*
+* *MANUAL*
+* *DUPS_OK*
+
+When you set the session acknowledgment to *IMMEDIATE*, the message gets acknowledged as soon as it's received. Do no mix *IMMEDIATE* acknowledgment with *Ack* or *Recovery session* operations.
+
+When you set the session acknowledgment to *AUTO*, the *On New Message* source takes control of the acknowledgment when the event finishes, or the session recovers when an error throws while processing the event. Adding a manual acknowledgment or session recover in an application with *AUTO* acknowledgment leads to duplicate operations on a channel.
+
+When you set the session acknowledgment to either *MANUAL* or *DUPS_OK*, check that every path of the flow execution gets a message acknowledgment or a session recover. Otherwise, a message can leak and lead to an unresponsive consumer once the max depth of the unit of work is reached.
+
+To resolve this issue:
+
+Take a detailed look at the flow design and be sure that every message gets acknowledged or the session recovers.
+
+== Troubleshoot MQ Stopped Consuming Messages
+
+Usually when MQ stopped consuming messages is related to a large number of unacknowledged messages. When you have a Mule app that has an *On New Message* source or *Consume* operation with manual acknowledgement of messages and not every path of the message has the proper acknowledgement or session recover, the Mule app can lead to an unresponsive consumer.
+
+To resolve this error:
+
+Check the depth of unacknowledged messages on the broker side and review the application logic.
```

Comment:
Do we have two separate lines for this element because in some cases the resolution requires more than one sentence? It's unnecessary in this topic set, but maybe is in the template for other situations?